Bind more functions for Vec2 and Rect

Gadgetoid opened the pull request:
Additionally add lua_blit_push<type> for existing types.

Daft-Freak committed:
pushpen helper

Daft-Freak committed:
Attempt to wrap file open mode

Daft-Freak committed:
More stdio wrappers/stubs

Daft-Freak committed:
watermark/now bindings

Daft-Freak committed:
Sprite overloads

Gadgetoid committed:
Bind more functions for Vec2 and Rect

Additionally add lua_blit_push<type> for existing types.

Gadgetoid committed:
Use asset pipeline for dingbads.bin

Gadgetoid committed:
Add Timer support and basic text function

Gadgetoid committed:
Expand text function

* Add font support

Co-authored-by: Charlie Birks <charlie@daft.games>

Daft-Freak commented:
Guess you can drop https://github.com/32blit/32blit-lua/pull/5/commits/acd08d4168469923114a4175c08c03cad41a5ad7?

Gadgetoid committed:
Text example and fixes

Gadgetoid committed:
Fixes, improvements, closer to 32Blit C++ API

This set of changes brings the Lua bindings closer in line with the 32Blit C++ API, for better or worse.

* All screen related functions in the "blit" namespace have been moved to "screen" instead
* Screen functions that should be properties are now properties, ie:
  * `blit.pen(Pen())` is now `screen.pen = Pen()`
  * `blit.clip(Rect())` is now `screen.clip = Rect()`
* screen.cpp created to keep all the screen stuff together
* sprites.cpp is currently unused since these are functions on "screen"
* "screen" is no longer a module loaded by main.cpp" but a table + metatable combo that allows a mix of functions/properties
* snake and text examples updated for API changes

Gadgetoid committed:
Add sprite-text example, updated for API changes

Co-authored-by: Charlie Birks <charlie@daft.games>

Gadgetoid commented:
> Guess you can drop [acd08d4](https://github.com/32blit/32blit-lua/commit/acd08d4168469923114a4175c08c03cad41a5ad7)?

Probably! I need to go through and prune all of these!

I've just broken the whole API better... :laughing:

Gadgetoid committed:
Update main.lua to match API

Gadgetoid committed:
Drop unecessary argument pops from bindings

Gadgetoid committed:
Complete text example, minus custom font

The C++ text example, ported more or less verbatim to Lua.

Small fix to sprite-test.lua for now() which had been moved to "blit"

Gadgetoid committed:
Add Vec3 and operators

Introduces Vec3, albeit there's no demonstration of it yet.

Additionally adds operators for Size, Point and Vec2 and updates snake.lua to take advantage of them.

This results in much cleaner code.
